File: LogiOS/TrustarApp/Repo/KyuyuRepo.swift
//
//  KyuyuRepo.swift
//  TrustarApp
//
//  Created by CuongNguyen on 03/03/2022.
//  Copyright © 2022 DionSoftware. All rights reserved.
//

import Foundation

class KyuyuRepo {
    private var resultDb: ResultDb
    private var user: LoggerUser
    
    init(_ resultDb: ResultDb, _ user: LoggerUser) {
        self.resultDb = resultDb
        self.user = user
    }
    
    func refuel(refueled: Refueled) {
        let k = CommonKyuyu(userInfo: user.userInfo, refueled: refueled, location: Current.Shared.lastLocation)
        resultDb.commonKyuyuDao?.insert(commonKyuyu: k)
    }

    func totalAmountOfCompany(allocationNo: String, fuelClass: String) -> Double {
        let userInfo = user.userInfo
        return resultDb.commonKyuyuDao?
                .getTotalAmount(companyCd: userInfo.companyCd, allocationNo: allocationNo, fuelClassCd: fuelClass) ?? 0
    }
}

File: LogAndroid/TrustarApp/src/main/kotlin/jp/co/toukei/log/trustar/repo/KyuyuRepo.kt
package jp.co.toukei.log.trustar.repo

import io.reactivex.rxjava3.core.Flowable
import jp.co.toukei.log.trustar.Current
import jp.co.toukei.log.trustar.db.result.entity.CommonKyuyu
import jp.co.toukei.log.trustar.other.Refueled
import jp.co.toukei.log.trustar.user.LoggedUser
import java.util.*

class KyuyuRepo(private val user: LoggedUser) {
    private val resultDb = user.resultDb
    fun refuel(refueled: Refueled) {
        val k = CommonKyuyu(
            user.userInfo,
            refueled,
            Current.lastLocation
        )
        resultDb.commonKyuyuDao().insert(k)
    }

    fun totalAmountOfCompany(allocationNo: String, fuelClass: String): Flowable<Optional<Float>> {
        val userInfo = user.userInfo
        return resultDb.commonKyuyuDao().getTotalAmount(userInfo.companyCd, allocationNo, fuelClass)
    }
}
